### ISSUE TITLE
Allow skipping sudo password dialog

### ISSUE THREAD

**Issue by pktpls** [maxgallup/tailscale-status#12]
Tailscale now allows to execute `tailscale up` and other stuff as a regular user. This can be enabled with the `--operator` flag: `tailscale up --operator=$USER`

I think it might be worth for the extension to try commands without sudo first, and only show the password dialog if the command fails with a permission error. Or some other way of detecting that `--operator` is enabled.

**Comment by maxgallup** [comment]
Unfortunately I've been running into issues with trying to implement this feature. Running `tailscale up --operator=$USER` the first time will require to be run with sudo. After that you are able to run `tailscale up/down` as a non root user. 

This means that the extension would have to prompt for the password the first time using `pkexec` instead of `sudo`. The problem then seems to be that after prompting for the password the first time, it doesn't seem to register the --operator=$USER aspect and then requiring the password every time.

The following command illustrates what I mean:

`tailscale up --operator=$USER || pkexec tailscale up --operator=$USER`

If the first tailscale commands fails it will prompt for the password and try again using pkexec. However subsequent runs of this command seem to always prompt for the password within the extension, even though this command works perfectly if it's just typed into the terminal. 

Ideal solution would be to access tailscale's localapi directly (instead of running taiscale up/down commands), however there is missing documentation - suggestions or help appreciated!

**Comment by Aiq0** [comment]
I have experimented with it and it seems like `Gio.Subprocess.new` is not using shell to evaluate command, so `||` or `$USER` won't work, so you we need to call it 2 times (once without pkexec and if it is not successful, then with pkexec) and get username from `Glib.get_user_name()` and then, it seems to work perfectly fine.

But there is one caveat, if you specify some option (like --operator) in up command and you have set some options, you get this:
```
Error: changing settings via 'tailscale up' requires mentioning all
non-default flags. To proceed, either re-run your command with --reset or
use the command below to explicitly mention the current value of
all non-default settings:

	tailscale up --operator=my_username --shields-up
```
So we can either call it with `--reset` or get arguments from error message.

My solution was to do it like that and it seems to work perfectly fine:
```js
function cmdTailscale(args, unprivileged = true) {
    let command = (unprivileged ? ["tailscale"] : ["pkexec", "tailscale"]).concat(args);

    try {
        let proc = Gio.Subprocess.new(
            command,
            Gio.SubprocessFlags.STDOUT_PIPE | Gio.SubprocessFlags.STDERR_PIPE
        );
        proc.communicate_utf8_async(null, null, (proc, res) => {
            try {
                proc.communicate_utf8_finish(res);
                if (!proc.get_successful()) {
                    if (unprivileged) {
                        log("trying priviledged");
                        cmdTailscale(args[0] == "up" ? args.concat(["--operator=" + GLib.get_user_name(), "--reset"]) : args, false)
                    } else {
                        log(args);
                        log("failed @ cmdTailscale");
                    }
                } else {
                    cmdTailscaleStatus()
                }
            } catch (e) {
                logError(e);
            }
        });
    } catch (e) {
        logError(e);
    }
}
```
So basically it tries to run command without pkexec, and if it fails, it tries it with pkexec (and if it is `up` command, it sets `--operator` and `--reset` flags).

Same approach can be used for sending and receiving files without need of pkexec (with fallback to it).

(If you want, I can put it into pull request)

**Comment by maxgallup** [comment]
Very nice, thank you for the solution.
